### DISCUSSION TITLE
Attach to existing process?

### DISCUSSION THREAD

**Discussion by thenige** [Tyrrrz/CliWrap#173] [Q&A] [answered]
Hello, I see in the Command class there is an AttachAsync method that is private.  Is there any way to attach to an already running process?  

My use case is that I have a manager application that initially spawns long running console applications, but the manager may be closed by the user at any time and when they re-open it, the manager needs to re-connect to those already running processes that it already started previously.

**Comment by Tyrrrz** [comment, marked as answer]
There is no way because there's just too many things CliWrap needs to configure on `ProcessStartInfo` in order to work correctly. The `AttachAsync` method is private because it's used in a context where we attach to a process already launched by CliWrap.

**Comment by thenige** [comment]
Thanks for the answer. I was hoping to make use CliWrap for this - I'll continue managing these started processes manually.
